src/coreclr/src/vm/readytoruninfo.cpp

```diff
@@ -706,9 +711,9 @@ ReadyToRunInfo::ReadyToRunInfo(Module * pModule, PEImageLayout * pLayout, READYT
     }
 
     // For format version 4.1 and later, there is an optional inlining table
-    if (IsImageVersionAtLeast(4, 1))
+    if (pModule != NULL && IsImageVersionAtLeast(4, 1))
```

Comment:
We need comment on the `m_pModule` field about what situations can it be null.

Also, there is a ton of other places in this file where `m_pModule` is used. Are all of them safe? It feels like a bug farm to allow `m_pModule` to be null.

Reply:
I believe the case is relatively well isolated but I agree we should have stronger means to enforce this, I'll try to figure out some type hierarchy representing this. The case where m_pModule is null only happens for the "pseudo-ReadyToRunInfo" representing the composite image as such (as contrast to ReadyToRunInfo instances representing its component assemblies). It shouldn't leak anywhere into other runtime code and is only internally used by the "component-level" ReadyToRunInfo instances.